src/test/java/org/nuxeo/binary/metadata/test/BaseBinaryMetadataTest.java

```diff
@@ -0,0 +1,40 @@
+package org.nuxeo.binary.metadata.test;
+
+import static org.junit.Assert.assertFalse;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.nuxeo.binary.metadata.api.BinaryMetadataService;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.platform.commandline.executor.api.CmdParameters;
+import org.nuxeo.ecm.platform.commandline.executor.api.CommandLineExecutorService;
+
+/**
+ * @since 11.1
+ */
+public abstract class BaseBinaryMetadataTest {
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected BinaryMetadataService binaryMetadataService;
+
+    @Inject
+    protected CommandLineExecutorService commandLineExecutorService;
+
+    protected void itShouldFailIfOriginalFileIsCreatedByExifTool() throws IOException {
```

Comment:
Wouldn't it be easier to do it in a `@After` method instead of calling it in each test?

Reply:
+1 it makes sense as a check of resources disposal after work

Reply:
I thought of doing it, using the @After or at least just doing one check at the end of Test. But unfortunately some tests, try to write exift meta data more then one time in the same test and i need to be sure that after each call to the write method, the temporay directory doesn't hold any _original file

org.nuxeo.binary.metadata.test.TestBinaryMetadataOperation#itShouldWriteMetadataOnBinary

Others tests call more then one time the save method ( session.saveDocument)  and this one  will dispatch an event which is resulting on the call of Write method, two times. This why i recheck after each save 

org.nuxeo.binary.metadata.test.TestBinaryMetadataSyncListener#testEXIFandIPTC

for the others methods, there is one call, we can replace it by @After, but i didn't do it, just to stay coherant with others calls :

- The two methods where we can use @After
org.nuxeo.binary.metadata.test.TestBinaryMetadataService#itShouldAcceptQuoteInMetadataAndAllASCII

org.nuxeo.binary.metadata.test.TestBinaryMetadataSyncListener#testCollaborativeSaveOnlyIncrementsVersionOnce